helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml

```diff
@@ -166,6 +166,9 @@ starrocksFESpec:
 
 # spec for compute node, compute node provide compute function.
 starrocksCnSpec:
+  # number of replicas to deploy for cn component.
+  # we should not set default replicas, because if hpa is enabled, the replicas should be unset.
```

Comment:
The operator will eliminate replicas from the CN Statefulset.
However, if the HPA section is removed from the CN Spec, the default replicas will come into play, which is undesirable.

Reply:
then what's the exact behavior?

If `replica` is provided, it will take precedence over HPA settings?

Reply:
In the implementation of the operator: when both the replica and HPA are set in the CN Spec, the replicas field of the cn statefulset created by the operator will be set to null. This ensures that the replicas field is controlled by the HPA. Next, if a user's deployment, for example, does not involve changes to CN, then the Pod will not be deleted and recreated. However, if the user removes the hpa part from the CN Spec, the corresponding hpa object should be deleted, and if the replicas in the CN Spec still remain null, the pod will not be recreated.

If we set a default value for replicas in the chart, once the user removes the hpa part of the CN Spec, the number of pods will immediately revert to the replicas count, even though the user did not specify the replicas count in their values.yaml.

Reply:
it is fine we don't set the default value for the replica, it is the implementation detail that when the user's values provides both `replica` and `hpa` setting, what the exact behavior will be.

And we need to describe the accurate behavior, right now the statement `because if hpa is enabled, the replicas should be unset.` is missleading.